pkg/util/containers/v2/metrics/collector_ecs_fargate.go

```diff
@@ -0,0 +1,193 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+// +build docker
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/DataDog/datadog-agent/pkg/util/ecs/metadata"
+	v2 "github.com/DataDog/datadog-agent/pkg/util/ecs/metadata/v2"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	ecsFargateCollectorID = "ecs_fargate"
```

Comment:
Could it a generic ECS collector, set to priority 1 (not used in ECS EC2 except if system not available?)

Reply:
Turned out to be impossible as on ecs ec2 we can't list other task stats